src/Microsoft.AspNet.Mvc.WebApiCompatShim/ParameterBinding/IOptionalBinderMetadata.cs

```diff
@@ -4,13 +4,13 @@
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     /// <summary>
-    /// An <see cref="IBinderMetadata"/> that designates an optional parameter for the purposes
+    /// An type that designates an optional parameter for the purposes
     /// of WebAPI action overloading. Optional parameters do not participate in overloading, and 
     /// do not have to have a value for the action to be selected.
     /// 
     /// This has no impact when used without WebAPI action overloading.
     /// </summary>
-    public interface IOptionalBinderMetadata : IBinderMetadata
+    public interface IOptionalBinderMetadata
```

Comment:
Is this used anywehre? There should be a case in the model builder that checks for it, and stores the value in the new hashset you added.